sequence_task_1/student/include/impl.h

```diff
@@ -20,31 +21,77 @@ class StaticPlaylist
 {
 public:
     /** @todo Member traits */
+    typedef typename Container::iterator iterator;
+    typedef Song value_type;
+    typedef Song& reference;
+    typedef const Song& const_reference;
+    typedef typename Container::const_iterator const_iterator;
+    typedef std::ptrdiff_t difference_type;
+    typedef std::size_t size_type;
 
     /** @todo Iterators */
+    const_iterator begin() const
+    {
+        return m_tracklist.begin();
+    }
+
+    const_iterator end() const
+    {
+        return m_tracklist.end();
+    }
 
     /** @todo Constructor from any reversible sequence container */
+    template<typename Cont>
+    StaticPlaylist(Cont cont)
+        : m_tracklist(cont.begin(), cont.end())
+    {
+    }
 
     /** @todo Assignment from any reversible sequence container */
 
     /** @todo Add track from initializer */
     template<class... Args>
-    const Song& play(Args&&... songData);
+    const Song& play(Args&&... songData)
+    {
+        m_tracklist.emplace(m_tracklist.begin(), std::forward<Song>(Song{std::string{songData}})...);
```

Comment:
It would be very ineffective operation for vector!
Hint: what if we will reverse input container?